tests/app/main/views/test_suppliers.py

```diff
@@ -231,47 +213,79 @@ def test_should_raise_http_error_from_api(self):
         response = self.client.get('/admin/suppliers')
         assert response.status_code == 404
 
-    def test_should_list_suppliers(self):
+    def test_should_list_suppliers_with_pagination(self):
         self.data_api_client.find_suppliers.return_value = {
             "suppliers": [
-                {"id": 1234, "name": "Supplier 1"},
-                {"id": 1235, "name": "Supplier 2"},
-            ]
+                SupplierStub(id=1234, name="Supplier 1").response(),
+                SupplierStub(id=1235, name="Supplier 2").response()
+            ],
+            "links": {
+                'prev': 'http://localhost/suppliers?page=1&name=foo',
+                'self': 'http://localhost/suppliers?page=2&name=foo',
+                'next': 'http://localhost/suppliers?page=3&name=foo',
+                'last': 'http://localhost/suppliers?page=99&name=foo',
+            }
         }
         self.data_api_client.get_supplier_framework_info.side_effect = [
             {"frameworkInterest": {"agreementPath": "path/the/first/1234-g7-agreement.pdf"}},
             {"frameworkInterest": {"agreementPath": None}},  # Supplier 1234 has not returned their DOS agreement yet
             HTTPError(Response(404)),                        # Supplier 1235 is not on G-Cloud 7
             {"frameworkInterest": {"agreementPath": "path/the/third/1235-dos-agreement.jpg"}},
         ]
-        response = self.client.get("/admin/suppliers")
+        response = self.client.get("/admin/suppliers?supplier_name=foo&page=2")
         document = html.fromstring(response.get_data(as_text=True))
 
         assert response.status_code == 200
         assert len(document.cssselect('.summary-item-row')) == 2
 
+        page_content = response.get_data(as_text=True)
+        assert 'Previous page' in page_content
```

Comment:
I'd *personally* do these with a bit of xpath like `.xpath("//a[normalize-space(string())=$t]", t="Previous page")` and then yank the href out of the returned element, but it's not like any of the other tests around here do that.

Reply:
Agree - laziness on my part as there was nothing to easily copy and past 😸 Will add a commit to sharpen it up.